pkg/operator/ceph/cluster/osd/deviceSet.go

```diff
@@ -96,11 +96,17 @@ func (c *Cluster) createDeviceSetPVCsForIndex(deviceSet rookv1.StorageClassDevic
 
 	var dataSize string
 	var crushDeviceClass string
+	typesFound := util.NewSet()
 	for _, pvcTemplate := range deviceSet.VolumeClaimTemplates {
 		if pvcTemplate.Name == "" {
 			// For backward compatibility a blank name must be treated as a data volume
 			pvcTemplate.Name = bluestorePVCData
 		}
+		if typesFound.Contains(pvcTemplate.Name) {
+			errs.addError("found duplicate volume claim template %q for device set %q", pvcTemplate.Name, deviceSet.Name)
+			continue
+		}
+		typesFound.Add(pvcTemplate.Name)
```

Comment:
Is this intentional?

Reply:
The bug first surfaced as the missing metadata, so this check will help guard against incorrect metadata on the volumeClaimTemplate in the future. So it improves error checking, and the other commit fixes the underlying issue